plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java

```diff
@@ -0,0 +1,147 @@
+
+    @Inject
+    public TransportUpdateModelGroupAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        NamedXContentRegistry xContentRegistry
+    ) {
+        super(MLUpdateModelGroupAction.NAME, transportService, actionFilters, MLUpdateModelGroupRequest::new);
+        this.actionFilters = actionFilters;
+        this.transportService = transportService;
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<MLUpdateModelGroupResponse> listener) {
+        MLUpdateModelGroupRequest updateModelGroupRequest = MLUpdateModelGroupRequest.fromActionRequest(request);
+        MLUpdateModelGroupInput updateModelGroupInput = updateModelGroupRequest.getUpdateModelGroupInput();
+
+        String modelGroupId = updateModelGroupInput.getModelGroupID();
+        GetRequest getModelGroupRequest = new GetRequest(ML_MODEL_GROUP_INDEX).id(modelGroupId);
+
+        if (modelGroupId != null) {
+            User user = RestActionUtils.getUserContext(client);
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                client.get(getModelGroupRequest, ActionListener.wrap(modelGroup -> {
+                    if (modelGroup.isExists()) {
+                        Map<String, Object> source = modelGroup.getSourceAsMap();
+                        Map<String, Object> owner = (Map) source.get(MLModelGroup.OWNER);
+                        String access = (String) source.get(MLModelGroup.ACCESS);
+
+                        if (SecurityUtils.isAdmin(user) || access == "public") {
+                            updateModelGroup(modelGroupId, source, updateModelGroupInput, listener);
+                        } else if (user.getName().equals(owner.get("name"))) {
+                            if (updateModelGroupInput.getBackendRoles() != null) {
+
+                                List<String> backendRoles = user.getBackendRoles() != null ? user.getBackendRoles() : ImmutableList.of();
+
+                                if (updateModelGroupInput.getBackendRoles().size() == 0) {
+                                    updateModelGroupInput.setBackendRoles(backendRoles);
+                                }
+
+                                Boolean isRolePresent = updateModelGroupInput
+                                    .getBackendRoles()
+                                    .stream()
+                                    .allMatch(user.getBackendRoles().stream().collect(Collectors.toSet())::contains);
+
+                                if (!isRolePresent) {
+                                    log.error("Invalid Backend Roles provided in the input");
+                                    throw new IllegalArgumentException("Invalid Backend Roles provided in the input");
+                                }
```

Comment:
These code can be optimized to:
1. check `updateModelGroupInput.getBackendRoles() != null`.
2. if  `updateModelGroupInput.getBackendRoles().size() != 0`, check input backend roles is a subset of user backend roles. If false, throw exception.
3. else set user backend roles to updateModelGroupInput.
This can make the code more performant since the backend roles stream will run only when it's necessary.

Reply:
Actually made a change to this block of code since the default action for empty back-end roles is now private. Will push in my next commit in a while

Reply:
Another question is: don't we use addAllBackendRoles flag to determine if we add all the user's backend roles to the model?

Reply:
yes changing the code to accommodate all these scenarios